reference/nonfungible-token.scilla

```diff
@@ -128,83 +128,84 @@ transition transferSingle(to: ByStr20, tokenId: Uint256)
   end
 end
 
-(* @dev: Transfer the ownership of a given tokenId to another address *)
-(* @param: from    - Current owner address of the token               *)
-(* @param: to      - Recipient address for the token                  *)
-(* @param: tokenId - ID of the token to be transferred                *)
-transition transferFrom(from: ByStr20, to: ByStr20, tokenId: Uint256)
 
-  (* Get tokenOwner ByStr20 *)
-  getTokenOwner <- tokenOwnerMap[tokenId];
-  match getTokenOwner with
-  | None =>
-    (* Token not found *)
-    err = CodeNotFound;
-    MakeError err  
-  | Some tokenOwner =>
-    checkOwner = builtin eq tokenOwner _sender;
-    getApproved <- tokenApprovals[tokenId];
-    checkApproved =
-      match getApproved with
-      | Some app => builtin eq _sender app
-      | None => False
-      end;
-    getApprovedForAll <- operatorApprovals[tokenOwner][_sender];
-    checkApprovedForAll = 
-      match getApprovedForAll with 
-      | Some True => True 
-      | _ => False 
-      end;
-    (* Checks if the `from` is indeed the owner of the token *)
-    isFromTokenOwner = builtin eq tokenOwner from;
-    match isFromTokenOwner with
-    | False =>
-      (* From address is not the same as the tokenOwner    *)
-      err = CodeBadRequest;
-      MakeError err
-    | True => 
-      (* isApprovedOrOwner checks if any of the three conditions are met *)
-      isAuthorized = isApprovedOrOwner checkOwner checkApproved checkApprovedForAll;
-      match isAuthorized with
-      | True =>
-        (* Remove from Approval *)
-        match checkApproved with
-        | True =>
-          (* Remove entry from approvals at the token level *)
-          delete tokenApprovals[tokenId] 
+(* @dev Transfer the ownership of a given tokenId to another address  *)
+(* @param from:      Current owner of the token                       *)
+(* @param to:        Recipient address of the token                   *)
+(* @param tokenId:   uint256 id of the token to be transferred        *)
+transition transfer(from: ByStr20, to: ByStr20, tokenId: Uint256)
+    (* Get tokenOwner ByStr20 *)
+    getTokenOwner <- tokenOwners[tokenId];
+    match getTokenOwner with
+    | None =>
+        (* Token not found *)
+        err = CodeNotFound;
+        MakeError err  
+    | Some tokenOwner =>
+        checkOwner = builtin eq tokenOwner _sender;
+        getApproved <- tokenApprovals[tokenId];
+        checkApproved =
+          match getApproved with
+          | Some app => builtin eq _sender app
+          | None => False
+          end;
+        getApprovedForAll <- operatorApprovals[tokenOwner][_sender];
+        checkApprovedForAll = match getApprovedForAll with | Some True => True | _ => False end;
+
+        (* Checks if the `from` is indeed the owner of the token *)
+        isFromTokenOwner = builtin eq tokenOwner from;
```

Comment:
I still don't understand why `from` is necessary as a parameter.

In line 145 you check whether `_sender` is equal to `tokenOwner`, and if they are not equal the transition will fail with `CodeNotAuthorized`.

Here you check whether `from` is equal to `tokenOwner`, and if they are not equal the transition will fail with the not particularly descriptive `CodeBadRequest`.

Therefore, by transitivity of equality if `_sender` is not equal to `from`, then the transition will fail. Why not just use `_sender` instead of `from`, then?